feature/search/src/main/java/com/google/samples/apps/nowinandroid/feature/search/SearchViewModel.kt

```diff
@@ -17,14 +17,45 @@
 package com.google.samples.apps.nowinandroid.feature.search
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.samples.apps.nowinandroid.core.domain.GetSearchContentsUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
 import javax.inject.Inject
 
 @HiltViewModel
-class SearchViewModel @Inject constructor() : ViewModel() {
+class SearchViewModel @Inject constructor(
+    // TODO: Add GetSearchContentsCountUseCase to check if the fts tables are populated
+    getSearchContentsUseCase: GetSearchContentsUseCase,
+) : ViewModel() {
 
-    fun onSearchQueryChanged(searchQuery: String) {
-        // TODO: Pass it to UseCase
-        println("New search query is '$searchQuery'")
+    val searchQuery = MutableStateFlow("")
+
+    val searchResultUiState: StateFlow<SearchResultUiState> =
+        searchQuery.flatMapLatest { query ->
+            if (query.length < 2) {
+                flowOf(SearchResultUiState.EmptyQuery)
+            } else {
+                getSearchContentsUseCase(query).map {
```

Comment:
should handle error by `catch { emit(SearchResultUiState.Failure) }`

Reply:
Done